lib/jekyll.rb

```diff
@@ -96,7 +96,7 @@ def env
     #
     # Returns the final configuration Hash.
     def configuration(override = Hash.new)
-      config = Configuration[Configuration::DEFAULTS]
+      config = Configuration[Marshal.load(Marshal.dump(Configuration::DEFAULTS))]
```

Comment:
It makes a deep duplicate of the defaults. The PR body describes the problem of modifying the defaults in some situations.